File: backend/utils/logger.py
"""
Logger utility that conditionally logs based on environment
In production, all logging is disabled by default
"""
import os
from typing import Any


def _should_log() -> bool:
    """Check if logging should be enabled"""
    environment = os.getenv("ENVIRONMENT", "development").lower()
    enable_logging = os.getenv("ENABLE_LOGGING", "false").lower() == "true"

    # Enable logging if in development OR if explicitly enabled via env var
    return environment == "development" or enable_logging


class Logger:
    """Logger that respects environment settings"""

    def __init__(self):
        self.enabled = _should_log()

    def log(self, *args: Any, **kwargs: Any) -> None:
        """Print log message if logging is enabled"""
        if self.enabled:
            print(*args, **kwargs)

    def error(self, *args: Any, **kwargs: Any) -> None:
        """Print error message if logging is enabled"""
        if self.enabled:
            print("[ERROR]", *args, **kwargs)

    def warn(self, *args: Any, **kwargs: Any) -> None:
        """Print warning message if logging is enabled"""
        if self.enabled:
            print("[WARN]", *args, **kwargs)

    def info(self, *args: Any, **kwargs: Any) -> None:
        """Print info message if logging is enabled"""
        if self.enabled:
            print("[INFO]", *args, **kwargs)


# Create singleton instance
logger = Logger()

File: frontend/src/utils/logger.ts
/**
 * Logger utility that conditionally logs based on environment
 * In production, all logging is disabled by default
 */

const isDevelopment = import.meta.env.MODE === 'development';
const isLoggingEnabled = import.meta.env.VITE_ENABLE_LOGGING === 'true';

// Enable logging if in development OR if explicitly enabled via env var
const shouldLog = isDevelopment || isLoggingEnabled;

export const logger = {
  log: (...args: any[]) => {
    if (shouldLog) {
      console.log(...args);
    }
  },
  error: (...args: any[]) => {
    if (shouldLog) {
      console.error(...args);
    }
  },
  warn: (...args: any[]) => {
    if (shouldLog) {
      console.warn(...args);
    }
  },
  info: (...args: any[]) => {
    if (shouldLog) {
      console.info(...args);
    }
  },
};

// Default export for convenience
export default logger;
